File: app/src/main/java/andiag/coru/es/welegends/entities/Mastery.java
package andiag.coru.es.welegends.entities;

import java.io.Serializable;

/**
 * Created by Iago on 23/04/2015.
 */
public class Mastery implements Serializable {

    private long masteryId;
    private long rank;

    public Mastery() {
    }

    public Mastery(long masteryId, long rank) {
        this.masteryId = masteryId;
        this.rank = rank;
    }

    public long getRank() {
        return rank;
    }

    public void setRank(long rank) {
        this.rank = rank;
    }

    public long getMasteryId() {
        return masteryId;
    }

    public void setMasteryId(long masteryId) {
        this.masteryId = masteryId;
    }
}

File: WeLegends/app/src/main/java/es/coru/andiag/welegends/common/entities/Mastery.kt
package es.coru.andiag.welegends.common.entities

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.raizlabs.android.dbflow.annotation.*
import com.raizlabs.android.dbflow.structure.BaseModel
import es.coru.andiag.welegends.WeLegendsDatabase
import es.coru.andiag.welegends.models.utils.ConverterStringList
import java.io.Serializable

/**
 * Created by Canalejas on 13/12/2016.
 */
@Table(database = WeLegendsDatabase::class)
class Mastery : BaseModel(), Serializable {

    @Expose(serialize = false, deserialize = false) @PrimaryKey(autoincrement = true) var mid: Int = 0
    @SerializedName("id") @Unique @Column var riotId: Int? = null
    @Unique @Column var name: String? = null
    @ForeignKey(tableClass = Image::class) var image: Image? = null
    @Column var ranks: Int? = null
    @Column var prereq: String? = null
    @Column(typeConverter = ConverterStringList::class) var description: Array<String>? = null

}
